palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py

```diff
@@ -0,0 +1,301 @@
+        self.api_key_id = api_key_id
+        self.api_key = api_key
+        self.fully_qualified_domain_name = fully_qualified_domain_name
+        self.logger = logger
+        if security_level == "Advanced":
+            self.headers = self._advanced_authentication(self.api_key_id, self.api_key)
+        else:
+            self.headers = self._standard_authentication(self.api_key_id, self.api_key)
+
+    def test_connection(self):
+        endpoint = "/api_keys/validate/"
+        try:
+            self._post_to_api(endpoint, {})
+        except Exception as e:
+            raise ConnectionTestException(
+                cause="Connection Test Failed.",
+                assistance="Please check your connection settings and try again.",
+                data=e,
+            )
+
+    def get_endpoint_information(self, endpoint):
+        endpoint_type = self._get_endpoint_type(endpoint)
+        self.logger.info(f"Endpoint field to look for: {endpoint_type}")
+
+        api_endpoint = "/public_api/v1/endpoints/get_endpoint/"
+
+        request_body = {
+            "request_data": {
+                "search_from": 0,
+                "search_to": 100,
+                "sort": {"field": "endpoint_id", "keyword": "ASC"},
+                "filters": [{"field": endpoint_type, "operator": "in", "value": [endpoint]}],
+            }
+        }
+
+        self.logger.info("Posting to API...")
+        results = self._post_to_api(api_endpoint, request_body)
+        self.logger.info("Getting Reply...")
+
+        output_object = {
+            "endpoints": results.get("reply").get("endpoints"),
+            "total_count": results.get("reply").get("total_count"),
+        }
+
+        return output_object
+
+    def isolate_endpoint(self, endpoint, isolation_state):
+        endpoint_info = self.get_endpoint_information(endpoint)
+        endpoints = endpoint_info.get("endpoints")
+        if not endpoints:
+            self.logger.error(f"No matching endpoint found. Endpoint: {endpoint}")
+            raise PluginException(
+                cause="Endpoint not found",
+                assistance=f"The endpoint {endpoint} was not found. Please check your input and try again",
+            )
+
+        num_endpoints = len(endpoints)
+        self.logger.info(f"Number of endpoints found: {num_endpoints}")
+
+        if num_endpoints > 1:
+            self.logger.info(f"Taking isolation action on multiple endpoints.")
+            return self._isolate_multiple_endpoints(endpoints, isolation_state)
+        else:
+            self.logger.info(f"Taking isolation action on a single endpoint.")
+            return self._isolate_endpoint(endpoints, isolation_state)
+
+    def get_incidents(self, from_time, to_time, time_field="creation_time"):
+        endpoint = "/public_api/v1/incidents/get_incidents/"
+
+        batch_size = 100
+        search_from = 0
+        search_to = search_from + batch_size
```

Comment:
Should just set this to the value of `batch_size` since this will always start at that value given the two lines above